include/tbbexec/tbb_thread_pool.hpp

```diff
@@ -0,0 +1,435 @@
+            bulk_sender<stdexec::__x<std::remove_cvref_t<Sender>>, Shape, stdexec::__x<std::remove_cvref_t<Fun>>>;
+
+        template <stdexec::sender S, std::integral Shape, class Fn>
+        friend bulk_sender_t<S, Shape, Fn> tag_invoke(stdexec::bulk_t,
+                                                      const scheduler& sch,
+                                                      S&& sndr,
+                                                      Shape shape,
+                                                      Fn fun) noexcept {
+            return bulk_sender_t<S, Shape, Fn>{*sch.pool_, (S &&) sndr, shape, (Fn &&) fun};
+        }
+
+        friend stdexec::forward_progress_guarantee tag_invoke(stdexec::get_forward_progress_guarantee_t,
+                                                              const DerivedPoolType&) noexcept {
+            return stdexec::forward_progress_guarantee::parallel;
+        }
+
+        friend thread_pool_base;
+        explicit scheduler(DerivedPoolType& pool) noexcept : pool_(&pool) {}
+
+        DerivedPoolType* pool_;
+    };
+
+    [[nodiscard]] scheduler get_scheduler() noexcept { return scheduler{static_cast<DerivedPoolType&>(*this)}; }
+
+    /* Is this even needed? Looks like no.
+     * void request_stop() noexcept {
+        // Should this do anything? TBB supports its own flavor of cancelation at a tbb::task_group level
+        //
+    https://spec.oneapi.io/versions/latest/elements/oneTBB/source/task_scheduler/scheduling_controls/task_group_context_cls.html
+        // but not at the tbb::task_arena level.
+        // https://spec.oneapi.io/versions/latest/elements/oneTBB/source/task_scheduler/task_arena/task_arena_cls.html
+    }*/
+
+    [[nodiscard]] std::uint32_t available_parallelism() const { return static_cast<DerivedPoolType&>(*this).available_parallelism(); }
+
+private:
+    void enqueue(task_base* task, std::uint32_t tid = 0) noexcept { static_cast<DerivedPoolType&>(*this).enqueue(task, tid); }
+    void bulk_enqueue(task_base* task, std::uint32_t n_threads) noexcept {
+        for (std::size_t tid = 0; tid < n_threads; ++tid) {
+            this->enqueue(task, tid);
+        }
+    }
+};
+
+} // namespace detail
+
+
+class tbb_thread_pool : public detail::thread_pool_base<tbb_thread_pool> {
+public:
+    //! Constructor forwards to tbb::task_arena constructor:
+    template <typename... Args>
+    explicit tbb_thread_pool(Args&&... args) : arena_{std::forward<Args>(args)...} {
+        arena_.initialize();
+    }
+
+    [[nodiscard]] std::uint32_t available_parallelism() const { return arena_.max_concurrency(); }
+
+private:
+    friend detail::thread_pool_base<tbb_thread_pool>;
+
+    template <typename PoolType, typename ReceiverId>
+    friend class operation;
+
+    void enqueue(task_base* task, std::uint32_t tid = 0) noexcept {
+        arena_.enqueue([task, tid] { task->__execute(task, /*tid=*/tid); });
+    }
+
+    tbb::task_arena arena_{tbb::task_arena::attach{}};
+};
```

Comment:
... Now with `detail::thread_pool_base` pulled out it's clear how little actually depends on `tbb::task_arena`: There's a constructor, `available_parallelism()`, `void enqueue(task_base* task, std::uint32_t tid = 0)`, and the member.